web/ui/react-app/src/pages/graph/HistogramChart.tsx

```diff
@@ -5,19 +5,86 @@ import { bucketRangeString } from './DataTable';
 
 type ScaleType = 'linear' | 'exponential';
 
+type closestToZeroType = {
+  closest: number;
+  closestIdx: number;
+};
+
 const HistogramChart: FC<{ histogram: Histogram; index: number; scale: ScaleType }> = ({ index, histogram, scale }) => {
   const { buckets } = histogram;
+  const formatter = Intl.NumberFormat('en', { notation: 'compact' });
+
   const rangeMax = buckets ? parseFloat(buckets[buckets.length - 1][2]) : 0;
+  const rangeMin = buckets ? parseFloat(buckets[0][1]) : 0;
+
+  const expBucketWidth = buckets
+    ? Math.abs(
+        Math.log(Math.abs(parseFloat(buckets[buckets.length - 1][2]))) -
+          Math.log(Math.abs(parseFloat(buckets[buckets.length - 1][1])))
```

Comment:
If there are _only_ negative buckets, or even _only_ a zero bucket, this will break (just a note for the final code).